libarchive/archive_read_open_fd.c

```diff
@@ -185,7 +196,12 @@ static int64_t
 file_seek(struct archive *a, void *client_data, int64_t request, int whence)
 {
 	struct read_fd_data *mine = (struct read_fd_data *)client_data;
+#if defined(_WIN32) && !defined(__CYGWIN__)
+	/* We use _lseeki64() on Windows. */
+	int64_t seek = request;
+#else
 	off_t seek = (off_t)request;
+#endif
 	int64_t r;
 	int seek_bits = sizeof(seek) * 8 - 1;  /* off_t is a signed type. */
 
```

Comment:
If I remember the contracts correctly, `skip` is permitted to under-skip and the read core will recover, but `seek` must always handle the request exactly.  That means that the "reduce the request" step below might be a problem also.